File: Project Euler #12.cpp
#include <iostream>

int count_divs(int);

int main()
{
	bool sol = false;
	int num = 28 + 8, count = 8;
	while (!sol)
	{
		if (count_divs(num) >= 500)
			sol = true;
		else
		{
			count++;
			num += count;
		}
	}
	std::cout << num;
	std::cin.get();
	return 0;
}

int count_divs(int x)
{
	int sum = 2;
	for (int i = 2; i < x; i++)
	{
		if (x % i == 0)
			sum++;
	}
	return sum;
}

File: Python/Project Euler #12.py
sol = False
num = 28 + 8
count = 8
def count_divs(x):
    sum = 2
    for i in range(2, x):
        if x % i == 0:
            sum += 1
    return sum
while not sol:
    if count_divs(num) >= 500:
        sol = True
    else:
        count += 1
        num += count
print(num)
